src/odbc/src/statement.cpp

```diff
@@ -434,7 +439,7 @@ void Statement::SetAttribute(StatementAttributes& stmtAttr) {
   SetAttribute(SQL_ATTR_CONCURRENCY, &stmtAttr.concurrency, 0);
   SetAttribute(SQL_ATTR_CURSOR_TYPE, &stmtAttr.cursorType, 0);
   SetAttribute(SQL_ATTR_RETRIEVE_DATA, &stmtAttr.retrievData, 0);
-  SetAttribute(SQL_ATTR_ROW_ARRAY_SIZE, &stmtAttr.rowsetSize, 0);
+  SetAttribute(SQL_ROWSET_SIZE, &stmtAttr.rowsetSize, 0);
```

Comment:
I'm not familiar enough with this codebase to tell if this is correct, but it seems odd. What is this function for?

Reply:
This version of `SetStatementAttribute` is only used in one place, in `connection.cpp`:
```
SqlResult::Type Connection::InternalCreateStatement(Statement*& statement) {
  LOG_DEBUG_MSG("InternalCreateStatement is called");
  statement = new Statement(*this);

  if (!statement) {
    AddStatusRecord(SqlState::SHY001_MEMORY_ALLOCATION, "Not enough memory.");

    return SqlResult::AI_ERROR;
  }

  // set statement attributes that are set to this connection
  statement->SetAttribute(stmtAttr_);
  return SqlResult::AI_SUCCESS;
}
```

`stmtAttr_` has its fields set in `Connection::InternalSetStmtAttribute`.

This seems to be a way for `Connection` to use `Statement`. I changed the implementation so that the `rowsetSize` value is handled as `SQL_ROWSET_SIZE` `Statement::InternalSetAttribute`, instead of being treated as `SQL_ATTR_ROW_ARRAY_SIZE`.

Reply:
OK this makes sense. Based on the properties being set they look like they are the properties in ODBC 2.x SQLSetStmtOption function so it likely relates to that.